applications/StructuralMechanicsApplication/custom_conditions/line_load_condition.cpp

```diff
@@ -0,0 +1,458 @@
+/***********************************************************************************/
+
+template<std::size_t TDim>
+Condition::Pointer LineLoadCondition<TDim>::Create(
+    IndexType NewId,
+    NodesArrayType const& ThisNodes,
+    PropertiesType::Pointer pProperties
+    ) const
+{
+    return Kratos::make_intrusive<LineLoadCondition<TDim>>( NewId, GetGeometry().Create( ThisNodes ), pProperties );
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<std::size_t TDim>
+Condition::Pointer LineLoadCondition<TDim>::Clone (
+    IndexType NewId,
+    NodesArrayType const& ThisNodes
+    ) const
+{
+    KRATOS_TRY
+
+    Condition::Pointer p_new_cond = Kratos::make_intrusive<LineLoadCondition<TDim>>(NewId, GetGeometry().Create(ThisNodes), pGetProperties());
+    p_new_cond->SetData(this->GetData());
+    p_new_cond->Set(Flags(*this));
+    return p_new_cond;
+
+    KRATOS_CATCH("");
+}
+
+
+//******************************* DESTRUCTOR *****************************************
+/***********************************************************************************/
+
+template<std::size_t TDim>
+LineLoadCondition<TDim>::~LineLoadCondition()
+{
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<std::size_t TDim>
+void LineLoadCondition<TDim>::GetValueOnIntegrationPoints(
+    const Variable<array_1d<double, 3>>& rVariable,
+    std::vector< array_1d<double, 3>>& rOutput,
+    const ProcessInfo& rCurrentProcessInfo
+    )
+{
+    KRATOS_TRY;
+
+    this->CalculateOnIntegrationPoints( rVariable, rOutput, rCurrentProcessInfo );
+
+    KRATOS_CATCH( "" );
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<std::size_t TDim>
+void LineLoadCondition<TDim>::CalculateOnIntegrationPoints(
+    const Variable<array_1d<double, 3>>& rVariable,
+    std::vector< array_1d<double, 3>>& rOutput,
+    const ProcessInfo& rCurrentProcessInfo
+    )
+{
+    KRATOS_TRY;
+
+    const auto& r_geometry = this->GetGeometry();
+    const GeometryType::IntegrationPointsArrayType& r_integration_points = r_geometry.IntegrationPoints();
+    const IntegrationMethod integration_method = IntegrationUtilities::GetIntegrationMethodForExactMassMatrixEvaluation(r_geometry);
+
+    if ( rOutput.size() != r_integration_points.size() )
+        rOutput.resize( r_integration_points.size() );
+
+    if (rVariable == NORMAL) {
+        // Declaring tangent and Jacobian
+        array_1d<double, 3> tangent_xi, tangent_eta;
+        Matrix J(TDim, 1);
+
+        // Getting LOCAL_AXIS_2
+        GetLocalAxis2(tangent_eta);
```

Comment:
in any case this LocalAxis might change along the beam (think of a quadratic beam!)

Reply:
This is something you need to ask to @armingeiser, I don't know what it is supposed/intended/thought for this case

Reply:
Not sure if this is still relevant, but I am not involved a lot in beams and line loads, @loumalouomega maybe you meant someone else?

Reply:
@KlausBSautter

Reply:
Yes, sorry, my bad